v3/src/components/graph/models/graph-data-configuration-model.ts

```diff
@@ -414,10 +414,11 @@ export const GraphDataConfigurationModel = DataConfigurationModel
     })
   }))
   .views(self => ({
-    casesInRange(min: number, max: number, attrId: string, cellKey: Record<string, string>) {
+    casesInRange(min: number, max: number, attrId: string, cellKey: Record<string, string>, inclusiveMax = true) {
       return self.subPlotCases(cellKey)?.filter(caseId => {
         const caseValue = self.dataset?.getNumeric(caseId, attrId)
-        if (isFiniteNumber(caseValue) && caseValue >= min && caseValue <= max) {
+        const upperRangeBoundary = inclusiveMax ? max : max * 0.999999
```

Comment:
I think that rather than choosing an arbitrary multiplier (0.999999), it would be better to change the comparison.
const isLessThan = inclusiveMax ? caseValue <= upperRangeBoundary : caseValue < upperRangeBoundary
That way you don't have problems when values are very close to an upper boundary.